packages/financial-templates-lib/src/price-feed/FundingRateMultiplierPriceFeed.js

```diff
@@ -0,0 +1,89 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { BlockFinder } = require("./utils");
+const { ConvertDecimals } = require("@uma/common");
+const assert = require("assert");
+
+class FundingRateMultiplierPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new price feed object that tracks the funding rate multiplier in a particular perpetual contract.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} perpetualAbi Perpetual abi object to create a contract instance.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {String} perpetualAddress Ethereum address of the perpetual to monitor.
+   * @param {Function} getTime Returns the current time.
+   * @param {Function} [blockFinder] Optionally pass in a shared blockFinder instance (to share the cache).
+   * @param {Integer} [minTimeBetweenUpdates] Minimum amount of time that must pass before update will actually run
+   *                                        again.
+   * @param {Integer} [priceFeedDecimals] Precision that the caller wants precision to be reported in.
+   * @return None or throws an Error.
+   */
+  constructor({
+    logger,
+    perpetualAbi,
+    web3,
+    perpetualAddress,
+    getTime,
+    blockFinder,
+    minTimeBetweenUpdates = 60,
+    priceFeedDecimals = 18
+  }) {
+    super();
+
+    // Assert required inputs.
+    assert(logger, "logger required");
+    assert(perpetualAbi, "perpetualAbi required");
+    assert(web3, "web3 required");
+    assert(perpetualAddress, "perpetualAddress required");
+    assert(getTime, "getTime required");
+
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.perpetual = new web3.eth.Contract(perpetualAbi, perpetualAddress);
+    this.uuid = `FundingRateMultiplier-${perpetualAddress}`;
+    this.getTime = getTime;
+    this.priceFeedDecimals = priceFeedDecimals;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.blockFinder = blockFinder || BlockFinder(web3.eth.getBlock);
+    this.convertDecimals = ConvertDecimals(18, priceFeedDecimals, web3);
```

Comment:
question: in what context would this _not_ be 18 decimals? does a perpetual with a non-18 decimal collateral still have a funding rate of 18 decimals? If it does, we should probs consider removing any notion of decimals from this feed to simplify it a bit.

Reply:
Yes, the value retrieved from the blockchain is always 18 decimals since it is a percentage. However, we still follow the pattern of returning the value in whatever decimals the user specifies. Should we be doing something different in this price feed?

Reply:
Right but the whole point of defining a `prifeFeedDecimals` is to match the value that the Price feed returns to the value that the DVM would return. If the DVM would scale a price (such as an old BTC identifier/USDC) then the price feed should scale it the same way.

In the case of this feed the DVM will always return the price scaled at 18 decimals and so I dont see a reason to let the user of the feed parameterize this as it will by default match up with what dvm returns at this scalling.

Reply:
I see no problem with enabling the flexibility for the user to decide.

Reply:
Yeah, but this price feed value isn't generally used alone. This output could be used directly or it could be used in some sort of aggregation where a particular decimal value is needed (fallback or medianizer price feeds). I think keeping it consistent with the other feeds is the safest option even if it makes this feed a little more complex.